### PR TITLE
Add support for elba cpld rtc

### PR DESCRIPTION
**Pull request by mushtaqk07**
Added support for cpld rtc driver in elba.

### CODE REVIEW

File: patch/5.10-pensando-add-rtc-cpld.patch [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+
+diff --git a/arch/arm64/boot/dts/pensando/elba-asic-common.dtsi b/arch/arm64/boot/dts/pensando/elba-asic-common.dtsi
+index d8ab370..b1ce2c0 100644
+--- a/arch/arm64/boot/dts/pensando/elba-asic-common.dtsi
++++ b/arch/arm64/boot/dts/pensando/elba-asic-common.dtsi
+@@ -62,11 +62,6 @@
+ 		reg = <0x4c>;
+ 	};
+ 
+-	rtc@51 {
```
Comment (ashwin-h) [inline patch/5.10-pensando-add-rtc-cpld.patch:24]: This is will affect the Ortano cards running Sonic(RTC won't work on Ortano if we remove this change)..We need to probably need to create new dts for Sonic on elba with lipari. and add the below entry there.

Comment (mushtaqk07) [reply patch/5.10-pensando-add-rtc-cpld.patch:24]: Ok, was not aware Sonic is supported on Ortano cards too. Let's discuss will schedule a meeting if we have to create new DTS file or if both rtc entries exist what will be the behavior?

File: patch/5.10-pensando-add-rtc-cpld.patch [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
++#include <linux/mfd/pensando-elbasr.h>
++
++#define RTC_BASE_OFFSET  0xA2
++/*
++ * Date/Time registers
++ */
++#define DT_100THS	    0x00
++#define DT_YEARS	    0x01
++#define DT_MONTHS   	0x02
++#define DT_DATE	        0x03
```
Comment (mushtaqk07) [inline patch/5.10-pensando-add-rtc-cpld.patch:131]: will correct the alignment.

Comment (mushtaqk07) [reply patch/5.10-pensando-add-rtc-cpld.patch]: addressed in latest push

File: patch/5.10-pensando-add-rtc-cpld.patch [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
++obj-$(CONFIG_RTC_DRV_ELBA_CPLD) += rtc-elbacpld.o
+ obj-$(CONFIG_RTC_DRV_ZYNQMP)	+= rtc-zynqmp.o
+diff --git a/drivers/rtc/rtc-elbacpld.c b/drivers/rtc/rtc-elbacpld.c
+new file mode 100644
+index 0000000..b578e4e
+--- /dev/null
++++ b/drivers/rtc/rtc-elbacpld.c
+@@ -0,0 +1,156 @@
++/*
++ * drivers/rtc/rtc-pcf85363.c
```
Comment (ashwin-h) [inline patch/5.10-pensando-add-rtc-cpld.patch:99]: Please correct the path.

File: patch/5.10-pensando-add-rtc-cpld.patch [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
++ *
++ * Driver for NXP PCF85363 real-time clock.
```
Comment (ashwin-h) [inline patch/5.10-pensando-add-rtc-cpld.patch:101]: Please change the description

File: patch/5.10-pensando-add-rtc-cpld.patch [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
++ *
++ * Copyright (C) 2017 Eric Nelson
```
Comment (ashwin-h) [inline patch/5.10-pensando-add-rtc-cpld.patch:103]: Please update the copyright.

File: patch/5.10-pensando-add-rtc-cpld.patch (hunk 1 above)
Comment (mushtaqk07) [reply patch/5.10-pensando-add-rtc-cpld.patch]: addressed by adding new dts  file for Lipari.

**Pull request merged**